TestBenches/InputRouter_test.cpp

```diff
@@ -546,75 +671,180 @@ int main(int argc, char * argv[])
+          , hInputStubs // input stub stream 
+          , hBx_o // output bx 
+          , hMemories);
+        break;
+      case 6 : 
+        InputRouterTop_IR_DTC_PS_1_B( hBx
+          , cLnkWrd // input link LUT 
+          , cPhBnWord  // n phi bins LUT 
+          , hInputStubs // input stub stream 
+          , hBx_o // output bx 
+          , hMemories);
+        break;
+      case 7 : 
+        InputRouterTop_IR_DTC_PS_2_A( hBx
+          , cLnkWrd // input link LUT 
+          , cPhBnWord  // n phi bins LUT 
+          , hInputStubs // input stub stream 
+          , hBx_o // output bx 
+          , hMemories); 
+        break;
+      case 8 : 
+        InputRouterTop_IR_DTC_PS_2_B( hBx
+          , cLnkWrd // input link LUT 
+          , cPhBnWord  // n phi bins LUT 
+          , hInputStubs // input stub stream 
+          , hBx_o // output bx 
+          , hMemories); 
+        break;
+      case 9 : 
+        InputRouterTop_IR_DTC_2S_1_A( hBx
+          , cLnkWrd // input link LUT 
+          , cPhBnWord  // n phi bins LUT 
+          , hInputStubs // input stub stream 
+          , hBx_o // output bx 
+          , hMemories); 
+        break;
+      case 10 : 
+        InputRouterTop_IR_DTC_2S_1_B( hBx
+        , cLnkWrd // input link LUT 
+        , cPhBnWord  // n phi bins LUT 
+        , hInputStubs // input stub stream 
+        , hBx_o // output bx 
+        , hMemories);
+        break;
+      case 11:
+        InputRouterTop_IR_DTC_2S_2_A( hBx
+        , cLnkWrd // input link LUT 
+        , cPhBnWord  // n phi bins LUT 
+        , hInputStubs // input stub stream 
+        , hBx_o // output bx 
+        , hMemories); 
+        break;
+      case 12:
+        InputRouterTop_IR_DTC_2S_2_B( hBx
+        , cLnkWrd // input link LUT 
+        , cPhBnWord  // n phi bins LUT 
+        , hInputStubs // input stub stream 
+        , hBx_o // output bx 
+        , hMemories);  
+        break;
+      case 13: 
+        InputRouterTop_IR_DTC_2S_3_A( hBx
+        , cLnkWrd // input link LUT 
+        , cPhBnWord  // n phi bins LUT 
+        , hInputStubs // input stub stream 
+        , hBx_o // output bx 
+        , hMemories); 
+        break;
+      case 14: 
+        InputRouterTop_IR_DTC_2S_3_B( hBx
+        , cLnkWrd // input link LUT 
+        , cPhBnWord  // n phi bins LUT 
+        , hInputStubs // input stub stream 
+        , hBx_o // output bx 
+        , hMemories); 
+        break;
+      case 15: 
+        InputRouterTop_IR_DTC_2S_4_A( hBx
+        , cLnkWrd // input link LUT 
+        , cPhBnWord  // n phi bins LUT 
+        , hInputStubs // input stub stream 
+        , hBx_o // output bx 
+        , hMemories); 
+        break;
+      case 16: 
+        InputRouterTop_IR_DTC_2S_4_B( hBx
+        , cLnkWrd // input link LUT 
+        , cPhBnWord  // n phi bins LUT 
+        , hInputStubs // input stub stream 
+        , hBx_o // output bx 
+        , hMemories); 
+        break;
+    } 
+       
```

Comment:
@sseifeln The IR currently fails C/RTL cosimulation because these lines cause a different top function to be called than what is specified in project/script_IR.tcl. Could you please fix?

Reply:
hi @aehart  - sure (sorry for the late response) - which top function should be called instead? I remember it was one of the PS10G IRs but not which one specifically..

Reply:
The default top function in project/script_IR.tcl is InputRouterTop_IR_DTC_PS10G_3_A, and that script passes a link ID of 8 to the test bench. That's unchanged from the master branch, but this switch statement, which came from the reduced_config_pr branch, seems to be inconsistent with those settings.

Reply:
ok thanks .. sorry we have a beam test on-going so didn't get around to it yet. But will make sure to commit a change tomorrow.

Reply:
@aehart  - just committed the changes and it now passes cosim : https://github.com/cms-L1TK/firmware-hls/commit/9c6438d2a5177ebcf931f59ab94ad2b9a386bb8b

Reply:
Thank you!